Source:
Internal

Info:
To take out buffer overflow issue.

Status:
Upstream seems not to bother about case 'str == NULL' like we do.
This is the only difference

--- libtasn1-4.13/lib/coding.c	2016-09-11 11:53:52.000000000 +0000
+++ libtasn1-4.13.modified/lib/coding.c	2018-01-23 11:06:40.168254599 +0000
@@ -338,7 +338,10 @@
   char *temp, *n_end, *n_start;
   unsigned char bit7;
   uint64_t val, val1 = 0;
-  int str_len = _asn1_strlen (str);
+  size_t str_len = (str == NULL) ? 0 : _asn1_strlen (str);
+
+  if ((str_len + 2) < 2)
+    return ASN1_MEM_ALLOC_ERROR;
 
   max_len = *der_len;
 
